test/widgets/lightbox_test.dart

```diff
@@ -197,6 +204,153 @@ class FakeVideoPlayerPlatform extends Fake
+        );
+        animatedPositions.add(tester.getRect(animatedFlightImageFinder));
+      }
+
+      final totalDistance = sqrt(
+        pow(initialImagePosition.top - animatedPositions.last.top, 2) +
+        pow(initialImagePosition.left - animatedPositions.last.left, 2));
+
+      Rect previousPosition = initialImagePosition;
+      double maxStepDistance = 0.0;
+      for (final position in animatedPositions) {
+        final stepDistance = sqrt(
+          pow(position.top - previousPosition.top, 2) +
+          pow(position.left - previousPosition.left, 2));
+        maxStepDistance = max(maxStepDistance, stepDistance);
+        check(position).not((pos) => pos.equals(previousPosition));
+
+        previousPosition = position;
+      }
+      check(maxStepDistance).isLessThan(0.03 * totalDistance);
+
+      debugNetworkImageHttpClientProvider = null;
+    });
+
+    testWidgets('no hero animation occurs between different message list pages for same image', (tester) async {
+      prepareBoringImageHttpClient();
+
+      await setupMessageListPage(tester);
+
+      final messageContentFinder = find.byWidgetPredicate((widget) =>
+        widget is MessageContent && widget.message.id == message.id
+      );
+      final firstImageFinder = find.descendant(
+        of: messageContentFinder,
+        matching: find.byType(RealmContentNetworkImage)
+      );
+      final imageSrcUrl = tester
+        .widget<RealmContentNetworkImage>(firstImageFinder).src;
+      final firstImagePosition = tester.getRect(firstImageFinder);
+      final firstElement = firstImageFinder.evaluate().single;
+
+      connection.prepare(json:
+        eg.newestGetMessagesResult(foundOldest: true, messages: [message]).toJson());
+
+      await tester.tap(find.descendant(
+        of: find.byType(StreamMessageRecipientHeader),
+        matching: find.text('test topic')));
+      await tester.pumpAndSettle();
+
+      final secondImageFinder = find.byWidgetPredicate((widget) =>
+        widget is RealmContentNetworkImage && widget.src == imageSrcUrl
+      );
+      final secondImagePosition = tester.getRect(secondImageFinder);
+      final secondElement = secondImageFinder.evaluate().single;
+
+      final backButton = find.byType(BackButton);
+      await tester.tap(backButton);
+      await tester.pump();
+
+      final heroAnimationDuration = const Duration(milliseconds: 300);
+      final steps = 150;
+      final stepDuration = heroAnimationDuration ~/ steps;
+      final firstImagePositions = <Rect>[];
+      final secondImagePositions = <Rect>[];
+
+      for (int i = 0; i < steps; i++) {
+        await tester.pump(stepDuration);
+
+        final currentImages = find.byWidgetPredicate((widget) =>
+          widget is RealmContentNetworkImage && widget.src == imageSrcUrl).evaluate();
+        check(currentImages)
+          ..length.equals(2)
+          ..contains(firstElement)
+          ..contains(secondElement);
```

Comment:
Fun shorthand for this:
```suggestion
        check(currentImages).unorderedEquals([firstElement, secondElement]);
```